src/librustdoc/clean/utils.rs

```diff
@@ -600,3 +600,36 @@ pub(super) fn display_macro_source(
         }
     }
 }
+
+crate fn primitive_from_str(path_str: &str) -> Option<PrimitiveType> {
```

Comment:
This is no longer used anywhere except intra-doc links, right? Can you revert this change? I don't think most places should be using strings to parse primitives.

Reply:
Ah right, moving it back then!